### PR TITLE
Add 16bit image/tiff file support

### PR DESCRIPTION
**Pull request by Cicero-Zhao**
for PIL cannot handle 16bit tiff file, so try to use opencv to convert 16bit tiff file to png file, and keep going on same process.

### CODE REVIEW

File: thumbor/engines/__init__.py [hunk 1]
```diff
@@ -117,6 +127,8 @@ def get_mimetype(cls, buffer):
             mime = 'video/mp4'
         elif buffer.startswith('\x1aE\xdf\xa3'):
             mime = 'video/webm'
+        elif buffer.startswith('\x49\x49\x2A\x00'):
```
Comment (kkopachev) [inline thumbor/engines/__init__.py]: Should also check for big endian:
```python
or buffer.startswith('\x4D\x4D\x00\x2A')
```

File: thumbor/engines/__init__.py [hunk 2]
```diff
@@ -147,6 +159,20 @@ def convert_svg_to_png(self, buffer):
...
+        if not numpy:
+            msg = """[BaseEngin] convert_tif_to_png failed numpy not imported"""
+            logger.error(msg)
+            return buffer
+        if not cv2:
+            msg = """[BaseEngin] convert_tif_to_png failed opencv not imported"""
+            logger.error(msg)
+            return buffer
+
+        img = cv2.imdecode(numpy.fromstring(buffer), -1)
```
Comment (kkopachev) [inline thumbor/engines/__init__.py]: `numpy.fromstring(buffer)` is complaining:
> string size must be a multiple of element size

So I tried setting second argument:
```python
numpy.fromstring(buffer, dtype='uint16')
```
and that works

**Comment by kkopachev** [comment]
Any update on this?

**Comment by guilhermef** [comment]
The tests are still missing.

**Comment by Cicero-Zhao** [comment]
Yes, sorry, I still can not find any easy method to test 16 bits tiff
format easily in unit test... I can revoke my commit and commit it again
after finding solution?

thank you and sorry again.

2016-06-04 19:17 GMT+08:00 Guilherme Souza <notifications@github.com>:

**Comment by guilhermef** [comment]
You can copy a base handler test, but with 16bit tiff.

**Comment by kkopachev** [comment]
@Cicero-Zhao I took your commit and added some fixes and tests. You can see it here: https://github.com/thumbor/thumbor/issues/746

**Comment by kkopachev** [comment]
This can be closed now, see #746. Thanks @Cicero-Zhao!

**Comment by Cicero-Zhao** [comment]
I see, I gained much skills from your commit, thank you very much!

**Pull request closed without merging**